src/hotspot/share/gc/shenandoah/shenandoahRegulatorThread.cpp

```diff
@@ -72,7 +72,7 @@ void ShenandoahRegulatorThread::regulate_concurrent_cycles() {
     ShenandoahControlThread::GCMode mode = _control_thread->gc_mode();
     if (mode == ShenandoahControlThread::none) {
       if (should_unload_classes()) {
-        if (_control_thread->request_concurrent_gc(GLOBAL)) {
+        if (_control_thread->request_concurrent_gc(GLOBAL_GEN)) {
```

Comment:
I'm not following when this new code triggers request_concurrent_GC(GLOBAL_NON_GEN)...

My question may relate to William's.

Reply:
Right. This is fixed in new commits.